package/src/frontend/VideoPlayer/mediaEvents.js

```diff
@@ -12,19 +12,21 @@ export const mediaEvents = function(player, context) {
     });
   }
 
-  player.on('play', function() {
-    triggerMediaEvent('play');
-  });
-
-  player.on('timeupdate', function() {
-    triggerMediaEvent('timeupdate');
-  });
-
-  player.on('pause', function() {
-    triggerMediaEvent('pause');
-  });
+  if (player.mediaEventsListener) {
+    player.off('play', player.mediaEventsListener);
+    player.off('timeupdate', player.mediaEventsListener);
+    player.off('pause', player.mediaEventsListener);
+    player.off('ended', player.mediaEventsListener);
+  }
 
-  player.on('ended', function() {
-    triggerMediaEvent('ended');
-  });
-};
\ No newline at end of file
+  if (context) {
+    player.mediaEventsListener = function (event) {
+      triggerMediaEvent(event.type);
+    }
+  
+    player.on('play', player.mediaEventsListener);
+    player.on('timeupdate', player.mediaEventsListener);
+    player.on('pause', player.mediaEventsListener);
+    player.on('ended', player.mediaEventsListener);
+  }
```

Comment:
This now only registers media events when context is passed. Before context was optional. I don't know if we depend on media events somewhere without passing a context. I think, if we introduce an `updateMediaEvents` method (similar to the already existing `updateHooks` method) these cases will be easier to separate.

Please add a spec for this now more complicated piece of functionality.

Reply:
Changed it to register media events for all players on creation. 
On allocation then media events context is updated.
players without media events context do not trigger media event.